### PR TITLE
[batch] Add resource ids to the attempt resources table

### PR DESCRIPTION
**Pull request by jigold**
This PR adds a new column to the `resources` table (resource_id) as well as to the `attempt_resources` table. The `resource_id` represents the mapping from the string resource name to an integer identifier. To make this migration work, we first need to add the new columns to the corresponding tables along with a new trigger. The trigger makes sure any writes that occur after we start populating the older values will have the new resource_ids filled into the `attempt_resources` table. I decided to not convert the `attempt_{batch, job,billing_project}_resources` tables to have the resource_id because we'll drop them anyways in the future and it was adding 75% more time to the migration. My original script had the updates happening to all tables, hence why there's overkill for the way it's structured. I can try and refactor the code back to the way it was before if you'd like.

The code to insert into the `attempt_resources` table in `job.py` needs to insert both the resource and the resource_id because we do not want to rely on the new trigger for adding the resource_id to the record so we can remove the resource column later on.

### CODE REVIEW

**Comment by jigold** [comment]
This is all working. I added back the randomization as its omission made performance substantially worse. I'm estimating it's going to take about 3 hours and adds ~50 GB to the database.

**Comment by jigold** [comment]
****************************************************************************************** 
**DO NOT MERGE BEFORE SWITCHING DATABASE TO 8 CORES IN GCP**
******************************************************************************************

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]
Functionality looks fine, just something missing from estimated-current I think.

File: batch/sql/estimated-current.sql
Comment (jigold) [reply batch/sql/estimated-current.sql:582]: Good catch!

**Comment by jigold** [comment]
Can I get an approval? I checked the WIP tag is still on, but it would be great if you could double check as well before approving.

**Review by daniel-goldstein — APPROVED** [review]

**Pull request merged**
